### PR TITLE
ReactOS in 2020

### PR DESCRIPTION
**Pull request by Extravert-ir**

### CODE REVIEW

File: content/project-news/reactos-in-2020.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+**Kernel-mode Driver Framework support** As a part of the storage stack switch, the KMDF was ported to ReactOS allowing more drivers to potentially be used in the OS. The port is not complete yet, but already allows to use some drivers.
+
+**Memory manager/Common cache subsystems improvements** This is done by Jérôme Gardou, in parallel with the PnP and storage work by Victor Perevertkin. These two subsystems (CC and MM) break records by both the amount of time spent and by the amount of people who worked on them. Current effort is aimed on better compatibility with Windows on running filesystem drivers. And there are quite some achievements here: ReactOS is finally able to work with `ntfs.sys` driver from Windows and can handle an open-source Microsoft `fastfat.sys` driver for FAT filesystem. The work is not finished yet, but is going to be merged in 2021 for sure.
+
+**Plug and Play subsystem improvements** A major refactoring is going on here, to make the driver compatibility more complete and improve the overall system stability. This already made possible for a long-awaited ability to boot from USB media in an off-the-tree branch
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:49]:
```suggestion
**Plug and Play subsystem improvements** A major refactoring is going on here, to make the driver compatibility more complete and improve the overall system stability. This already made possible for a long-awaited ability to boot from USB media in an off-the-tree branch.
```

File: content/project-news/reactos-in-2020.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Good old i386 still remains as our main platform, but the work on porting ReactOS to the other ones slowly goes on.
+
+* AMD64: most of the code was merged to master from the branch, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. Graphical interface still needs fixes
+* XBOX: the Original XBOX port of ReactOS is revived, and able to boot to the desktop
+* PC98: @disean started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities
+
+## Future work
+
+Several areas are being worked on currently, and will be continues in 2021. This includes:
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:67]:
```suggestion
Several areas are being worked on currently, and will be continued in 2021. This includes:
```

**Review by binarymaster — COMMENTED** [review]
Several improvements and fixes.

File: content/project-news/reactos-in-2020.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+## Shell changes
+
+Shell doesn't see much attention recently, due to most of the work concentrated in the kernel, but there are still some useful fixes and feature implementations.
+
+* Katayama Hirofumi MZ made a great job implementing the Filesystem change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes
+* Visual and usability improvements: "Size on disk" label in a file Properties dialog, key accelerators in Open/Save dialog and many more
+
+## RAPPS (a GSoC project)
+
+This is one of our three GSoC projects of the year. He Yang continues the work from previous GSoC years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots, command line scripting support.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:21]: UPD: now I got it, but here is suggestion to make it more clear.
```suggestion
This is one of our three GSoC projects of the year. He Yang continues the work done by the students from previous GSoC years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots, command line scripting support.
```

File: content/project-news/reactos-in-2020.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+title:       "ReactOS in 2020"
+author:      "Extravert34"
+date:        2020-12-28
+---
+
+Despite all the turbulence, we've got quite a rich year for ReactOS.
+Many bugs and instabilities were resolved, many more have been introduced. This year we hired two kernel developers full-time, this happens for the first time on the project history.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:8]: Please start a new sentence on a new line (without an empty line between them).
This makes it simpler to view a diff, and still renders as one paragraph.

File: content/project-news/reactos-in-2020.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+The post highlights some of the changes which may may be interesting to the community.
+
+## Shell changes
+
+Shell doesn't see much attention recently, due to most of the work concentrated in the kernel, but there are still some useful fixes and feature implementations.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:14]:
```suggestion
The Shell doesn't receive much attention as of recent, due to the fact that most of the work is concentrated in the kernel, but there are still some useful fixes and feature implementations.
```

File: content/project-news/reactos-in-2020.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (learn-more) [reply content/project-news/reactos-in-2020.md:21]: 'He continues on the work of others from previous GSoC years' or something along those lines would be clearer indeed.

File: content/project-news/reactos-in-2020.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+* Katayama Hirofumi MZ made a great job implementing the Filesystem change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes
+* Visual and usability improvements: "Size on disk" label in a file Properties dialog, key accelerators in Open/Save dialog and many more
+
+## RAPPS (a GSoC project)
+
+This is one of our three GSoC projects of the year. He Yang continues the work from previous GSoC years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots, command line scripting support.
+
+*picture*
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:23]: You can use the picture from the gsoc post, and even link to it:
img/blogs/gsoc2020-rapps-final-report.png

File: content/project-news/reactos-in-2020.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+## Compiler and tooling upgrade
+
+The aging RosBE 2.1 toolset has served us well since 2014, but it's time to move on and incorporate all the useful features implemented in modern compilers and build tools.
+We now have GCC 8.4, CMake 3.17 as our defaults in the new RosBE 2.2, while maintaining compatibility with the latest CMake. The transition didn't happen without flaws though, a lot of effort was put by Colin Finck to make modern CMake work on Windows 2003 (and thus on ReactOS because we maintain compatibility with it). All the patches are freely available in our GitHub.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:28]:
```suggestion
We now have GCC 8.4 and CMake 3.17 as our defaults in the new RosBE 2.2, while maintaining compatibility with the latest CMake.
The transition didn't happen without a few road bumps though, a lot of effort was put by Colin Finck to make modern CMake work on Windows Server 2003 (and thus on ReactOS because we maintain compatibility with it).
All the patches are freely available in our [GitHub](https://github.com/reactos/CMake/tree/cmake-3.17.2-reactos).
```

File: content/project-news/reactos-in-2020.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+Clang build is resurrected on the CI, with plans on supporting this compiler along with GCC and MSVC. ReactOS buils with Clang now, but still cannot boot though.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:30]:
```suggestion
Clang build is resurrected on the CI, with plans on supporting this compiler along with GCC and MSVC.
ReactOS builds with Clang now, however it is still unable to boot.
```

File: content/project-news/reactos-in-2020.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+A controversial decision was made to drop support of old MSVC compilers, and raise the C language standard to C99 for better cross-compiler compatibility. The deprecation brought up a big discussion over ML and GitHub.
+Still, this opens a window for future C++ standard library upgrades (maybe in 2021, who knows) and the support of modern C++ in our codebase.
+
+## Kernel and drivers
+
+As there are two developers working full-time in this field, there are quite some long-awaited changes:
+
+**Merge of the PnP storage stack** The new storage stack, derived from open-source Microsoft drivers came to replace the old NT4-compatible ReactOS storage stack. For the long time many workarounds were applied to make NT4 storage stack work in the NT5.2 environment. This era ends now, with a major step forward in compatibility with modern Windows, along with new features:
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:39]:
```suggestion
**Merge of the PnP storage stack** The new storage stack, derived from open-source Microsoft drivers came to replace the old NT4-compatible ReactOS storage stack.
For a long time many workarounds were applied to make the NT4 storage stack work in the NT5.2 environment.
This era ends now, with a major step forward in compatibility with modern Windows, along with new features:
```

File: content/project-news/reactos-in-2020.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+* GPT partitioning support
+* Support of the user-mode APIs related to storage stack
+* USB drives mount and dismount on-the-go
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:42]:
```suggestion
* USB drive mount and dismount on-the-go
```

File: content/project-news/reactos-in-2020.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+* Special SSD commands, Blu-ray disks, > 2TiB drives and many more
+
+**Kernel-mode Driver Framework support** As a part of the storage stack switch, the KMDF was ported to ReactOS allowing more drivers to potentially be used in the OS. The port is not complete yet, but already allows to use some drivers.
+
+**Memory manager/Common cache subsystems improvements** This is done by Jérôme Gardou, in parallel with the PnP and storage work by Victor Perevertkin. These two subsystems (CC and MM) break records by both the amount of time spent and by the amount of people who worked on them. Current effort is aimed on better compatibility with Windows on running filesystem drivers. And there are quite some achievements here: ReactOS is finally able to work with `ntfs.sys` driver from Windows and can handle an open-source Microsoft `fastfat.sys` driver for FAT filesystem. The work is not finished yet, but is going to be merged in 2021 for sure.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:47]:
```suggestion
**Memory manager/Common cache subsystems improvements** This is done by Jérôme Gardou, in parallel with the PnP and storage work by Victor Perevertkin.
These two subsystems (CC and MM) break records by both the amount of time spent and by the amount of people who worked on them.
Current effort is aimed on better compatibility with Windows on running filesystem drivers.
And there are quite some achievements here: ReactOS is finally able to work with the `ntfs.sys` driver from Windows and can handle an open-source Microsoft `fastfat.sys` driver for FAT filesystem.
The work is not finished yet, but is going to be merged in 2021 for sure.
```

File: content/project-news/reactos-in-2020.md (hunk 1 above)
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:49]:
```suggestion
**Plug and Play subsystem improvements** A major refactoring is going on here, to make the driver compatibility more complete and improve the overall system stability.
This already made it possible for the long-awaited ability to boot from USB media in an off-the-tree branch
```

File: content/project-news/reactos-in-2020.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+**Plug and Play subsystem improvements** A major refactoring is going on here, to make the driver compatibility more complete and improve the overall system stability. This already made possible for a long-awaited ability to boot from USB media in an off-the-tree branch
+
+All those changes wouldn't be possible without the donations from the community. Big thanks to all of the patrons!
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:51]:
```suggestion
All those changes wouldn't be possible without the donations from the community.
Big thanks to all of the donors!
```

'patrons' is now generally seen as a thing from the 'patreon' site, hence why I think it's better to change it to 'donors' instead, which is a more neutral term

File: content/project-news/reactos-in-2020.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+
+## CMD parser improvements and polishing
+
+_changes by hbelusca_
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:55]:
```suggestion
_changes by Hermès BÉLUSCA - MAÏTO_
```

Comment (learn-more) [reply content/project-news/reactos-in-2020.md:55]: You used names instead of aliases for other contributors as well ;)

Comment (learn-more) [reply content/project-news/reactos-in-2020.md:55]: oh, you are asking him to fill this in I guess?

File: content/project-news/reactos-in-2020.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+## Ports
+
+Good old i386 still remains as our main platform, but the work on porting ReactOS to the other ones slowly goes on.
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:59]:
```suggestion
Good old i386 still remains as our main platform, but the work on porting ReactOS to other architectures slowly goes on.
```

File: content/project-news/reactos-in-2020.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+* AMD64: most of the code was merged to master from the branch, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. Graphical interface still needs fixes
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:61]:
```suggestion
* AMD64: most of the code was merged to master, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. The graphical interface still needs fixes.
```

File: content/project-news/reactos-in-2020.md [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+* XBOX: the Original XBOX port of ReactOS is revived, and able to boot to the desktop
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:62]:
```suggestion
* XBOX: the Original XBOX port of ReactOS is revived, and able to boot to desktop.
```

File: content/project-news/reactos-in-2020.md [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+* PC98: @disean started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:63]:
```suggestion
* PC98: @disean started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities.
```
Might wanna include his name, since other contributors are named as well?
Possibly change all names to also link to their github profile

File: content/project-news/reactos-in-2020.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+
+## Future work
+
+Several areas are being worked on currently, and will be continues in 2021. This includes:
+
+* Forward compatibility subsystem for Windows Vista+ applications support
+* Printing
+* CRT and standard library improvements
+* The work on amd64 support and APIC HAL (Hardware Abstraction Layer) for the SMP support
```
Comment (learn-more) [inline content/project-news/reactos-in-2020.md:72]:
```suggestion
* Forward compatibility subsystem for Windows Vista+ applications support.
* Printing.
* CRT and standard library improvements.
* The work on amd64 support and APIC HAL (Hardware Abstraction Layer) for the SMP support.
```

File: content/project-news/reactos-in-2020.md [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+Despite all the turbulence, we've got quite a rich year for ReactOS.
+Many bugs and instabilities were resolved, many more have been introduced. This year we hired two kernel developers full-time, this happens for the first time on the project history.
+
+The post highlights some of the changes which may may be interesting to the community.
+
+## Shell changes
+
+Shell doesn't see much attention recently, due to most of the work concentrated in the kernel, but there are still some useful fixes and feature implementations.
+
+* Katayama Hirofumi MZ made a great job implementing the Filesystem change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:16]: + fullstop at end of sentence.

File: content/project-news/reactos-in-2020.md [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+* Visual and usability improvements: "Size on disk" label in a file Properties dialog, key accelerators in Open/Save dialog and many more
```
[the remark this answers is not in the document]
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:17]: ditto

File: content/project-news/reactos-in-2020.md (hunk 8 above)
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:30]: `The Clang build...`

File: content/project-news/reactos-in-2020.md (hunk 9 above)
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:39]: There is not a clear enough separation between the "titles" of the subparagraphs, and the text of these.
Maybe put the text of the paragraph on a newline below the "title"?

File: content/project-news/reactos-in-2020.md [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
+## Kernel and drivers
+
+As there are two developers working full-time in this field, there are quite some long-awaited changes:
+
+**Merge of the PnP storage stack** The new storage stack, derived from open-source Microsoft drivers came to replace the old NT4-compatible ReactOS storage stack. For the long time many workarounds were applied to make NT4 storage stack work in the NT5.2 environment. This era ends now, with a major step forward in compatibility with modern Windows, along with new features:
+* GPT partitioning support
+* Support of the user-mode APIs related to storage stack
+* USB drives mount and dismount on-the-go
+* Special SSD commands, Blu-ray disks, > 2TiB drives and many more
```
Comment (HBelusca) [inline content/project-news/reactos-in-2020.md:43]:
```suggestion
* GPT partitioning support;
* Support of the user-mode APIs related to storage stack;
* USB drive mount and dismount on-the-go;
* Special SSD commands, Blu-ray disks, > 2TiB drives,
* and many more...
```

File: content/project-news/reactos-in-2020.md (hunk 12 above)
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:51]: Yeah this sounded like the end of these YouTube Patreon-sponsored videos 😆

File: content/project-news/reactos-in-2020.md (hunk 16 above)
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:62]: Maybe thanking/mentioning @binarymaster and Daniel Reimer too because they helped in the port and the testing on XQEMU and real XBOXes.

File: content/project-news/reactos-in-2020.md (hunk 9 above)
Comment (learn-more) [reply content/project-news/reactos-in-2020.md:39]: You checked this in the page or the PR?
Because markdown renders quite different

Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:39]: I selected "view file" in order to have the markdown being rendered.

File: content/project-news/reactos-in-2020.md (hunk 13 above)
Comment (HBelusca) [reply content/project-news/reactos-in-2020.md:55]: @Extravert-ir : See my PM in mattermost.

File: content/project-news/reactos-in-2020.md (hunk 4 above)
Comment (ColinFinck) [inline content/project-news/reactos-in-2020.md:8]:
```suggestion
Many bugs and instabilities were resolved, many more have been introduced.
This year we hired two kernel developers full-time, which happens for the first time in project history.
```

File: content/project-news/reactos-in-2020.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Extravert-ir) [reply content/project-news/reactos-in-2020.md:21]: Yeah, I wanted to say this year is not the first one to have a RAPPS project

File: content/project-news/reactos-in-2020.md [hunk 22]
```diff
@@ -0,0 +1,83 @@
+---
+title:       "ReactOS in 2020"
+author:      "Extravert34"
+date:        2020-12-28
+---
+
+Despite all the turbulence, we've got quite a rich year for ReactOS.
```
Comment (Doug-Lyons) [inline content/project-news/reactos-in-2020.md:7]: Suggestion: Despite all the turbulence, it has been quite a productive year for ReactOS.

File: content/project-news/reactos-in-2020.md [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+Many bugs and instabilities were resolved, many more have been introduced.
+This year we hired two kernel developers full-time, which happens for the first time in project history.
```
Comment (Doug-Lyons) [inline content/project-news/reactos-in-2020.md:9]: Suggestion: ... which happened for... (past tense verb agreement)

File: content/project-news/reactos-in-2020.md [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+## Shell changes
+
+The Shell doesn't receive much attention as of recent, due to the fact that most of the work is concentrated in the kernel, but there are still some useful fixes and feature implementations.
+
+* Katayama Hirofumi MZ made a great job implementing the file system change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes
+* Visual and usability improvements: "Size on disk" label in File Properties dialog, key accelerators in Open/Save dialog and many more
+
+## RAPPS (a GSoC project)
+
+This is one of our three GSoC projects of the year. He Yang continues the work from previous GSoC years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots, command line scripting support.
```
Comment (Doug-Lyons) [inline content/project-news/reactos-in-2020.md:22]: Suggestion: Commonly the word "and" is included between the last comma and the final item such as:
"...showing app screenshots, and command line scripting support."

**Comment by Doug-Lyons** [comment]
Excellent write-up, @Extravert-ir. Thanks for your efforts.

File: content/project-news/reactos-in-2020.md [hunk 25]
```diff
@@ -0,0 +1,93 @@
...
+Still, this opens a window for future C++ standard library upgrades (maybe in 2021, who knows) and the support of modern C++ in our codebase.
+
+## Kernel and drivers
+
+As there are two developers working full-time in this field, there are quite some long-awaited changes:
+
+**Merge of the PnP storage stack** The new storage stack, derived from open-source Microsoft drivers came to replace the old NT4-compatible ReactOS storage stack.
+For a long time many workarounds were applied to make the NT4 storage stack work in the NT5.2 environment.
+This era ends now, with a major step forward in compatibility with modern Windows, along with new features:
+* GPT partitioning support.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:45]: This should fix bullet points in Hugo.
```suggestion
This era ends now, with a major step forward in compatibility with modern Windows, along with new features:

* GPT partitioning support.
```

File: content/project-news/reactos-in-2020.md [hunk 26]
```diff
@@ -0,0 +1,93 @@
...
+**Kernel-mode Driver Framework support** As a part of the storage stack switch, the KMDF was ported to ReactOS allowing more drivers to potentially be used in the OS. The port is not complete yet, but already allows to use some drivers.
+
+**Memory manager/Common cache subsystems improvements** This is done by Jérôme Gardou, in parallel with the PnP and storage work by Victor Perevertkin.
+These two subsystems (CC and MM) break records by both the amount of time spent and by the amount of people who worked on them.
+The current effort is aimed at better compatibility with Windows at running filesystem drivers.
+And there are quite some achievements here: ReactOS is finally able to work with the `ntfs.sys` driver from Windows and can handle an open-source Microsoft `fastfat.sys` driver for FAT filesystem.
+The work is not finished yet, but is going to be merged in 2021 for sure.
+
+**Plug and Play subsystem improvements** A major refactoring is going on here, to make the driver compatibility more complete and improve the overall system stability.
+This already made it possible for the long-awaited ability to boot from USB media in an off-the-tree branch
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:59]: Missing full stop
```suggestion
This already made it possible for the long-awaited ability to boot from USB media in an off-the-tree branch.
```

File: content/project-news/reactos-in-2020.md [hunk 27]
```diff
@@ -0,0 +1,93 @@
...
+A very big thanks to all of the donors!
+
+## CMD parser improvements and polishing
+
+The ReactOS Command Interpreter (CMD) was recently revisited, with a series of fixes aiming at improving its command and batch files parsing rules with those from Windows' CMD.
+This effort was triggered by the recent bug report [CORE-17030](https://jira.reactos.org/browse/CORE-17030) concering the incompatibility of our CMD to parse REM commands (comments that are ignored for the purposes of batch file execution),
+and by the fact that the overall command parsing of our CMD was not completely compatible with Windows' one,
+as per evidence with the many failures encountered during testing, with the `cmd:batch` Wine unit test (see left side of the picture below).
+
+HBM decided to make ReactOS' CMD parsing rules more compatible with Windows' one, including some obscure corner cases,
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:71]:
```suggestion
Hermès Bélusca-Maïto decided to make ReactOS' CMD parsing rules more compatible with Windows' one, including some obscure corner cases,
```

File: content/project-news/reactos-in-2020.md [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+and to take also the opportunity to bring additional fixes to some of its commands with the aim to fix more tests.
+
+Now, more `cmd:batch` tests are passing with overall less failures. He also added new tests for the CMD parser in our own `cmd:reactos` unit test. Most of these pass OK, except for some for which some extra work is still needed.
+
+HBM has also found that some aspecss of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:76]:
```suggestion
Hermès has also found that some aspecss of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
```

File: content/project-news/reactos-in-2020.md [hunk 29]
```diff
@@ -0,0 +1,93 @@
...
+
+All those changes wouldn't be possible without the donations from the community.
+A very big thanks to all of the donors!
+
+## CMD parser improvements and polishing
+
+The ReactOS Command Interpreter (CMD) was recently revisited, with a series of fixes aiming at improving its command and batch files parsing rules with those from Windows' CMD.
+This effort was triggered by the recent bug report [CORE-17030](https://jira.reactos.org/browse/CORE-17030) concering the incompatibility of our CMD to parse REM commands (comments that are ignored for the purposes of batch file execution),
+and by the fact that the overall command parsing of our CMD was not completely compatible with Windows' one,
+as per evidence with the many failures encountered during testing, with the `cmd:batch` Wine unit test (see left side of the picture below).
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:69]: > see left side of the picture below

Screenshot missing, cc @HBelusca

File: content/project-news/reactos-in-2020.md [hunk 30]
```diff
@@ -0,0 +1,93 @@
...
+
+HBM has also found that some aspecss of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
+
+## Ports
+
+Good old i386 still remains as our main platform, but the work on porting ReactOS to other architectures slowly goes on.
+
+* AMD64: most of the code was merged to master, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. The graphical interface still needs fixes.
+* XBOX: the Original XBOX port of ReactOS is revived, and able to boot to desktop.
+* PC98: @disean started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:84]:
```suggestion
* PC98: Dmitry Borisov started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities.
```

File: content/project-news/reactos-in-2020.md [hunk 31]
```diff
@@ -0,0 +1,93 @@
...
+Many bugs and instabilities were resolved, many more have been introduced.
+This year we hired two kernel developers full-time, this happened for the first time in the project's history.
+
+The post highlights some of the changes which may be interesting to the community.
+
+## Shell changes
+
+Shell hasn't seen much attention recently, due to most of the work being concentrated in the kernel, but there are still some useful fixes and feature implementations.
+
+* Katayama Hirofumi MZ made a great job implementing the file system change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:17]: Missing full stop
```suggestion
* Katayama Hirofumi MZ made a great job implementing the file system change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes.
```

File: content/project-news/reactos-in-2020.md [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+
+Now, more `cmd:batch` tests are passing with overall less failures. He also added new tests for the CMD parser in our own `cmd:reactos` unit test. Most of these pass OK, except for some for which some extra work is still needed.
+
+HBM has also found that some aspecss of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
+
+## Ports
+
+Good old i386 still remains as our main platform, but the work on porting ReactOS to other architectures slowly goes on.
+
+* AMD64: most of the code was merged to master, so now nightly builds can finally boot to the 1st stage setup screen on some hardware (Timo Kreuzer). The graphical interface still needs fixes.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:83]:
```suggestion
* AMD64: most of the code was merged to master, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. The graphical interface still needs fixes. The porting is being done by Timo Kreuzer.
```

File: content/project-news/reactos-in-2020.md [hunk 33]
```diff
@@ -0,0 +1,108 @@
...
+
+* Katayama Hirofumi MZ made a great job implementing the file system change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes.
+* Visual and usability improvements: "Size on disk" label in File Properties dialog, key accelerators in Open/Save dialog and many more.
+
+## RAPPS (a GSoC project)
+
+This is one of our three GSoC projects of the year. He Yang (kernel.bin) continues on the work made by GSoC students in previous years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots and command line scripting support.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/gsoc2020-rapps-final-report.png" src="/img/blogs/gsoc2020-rapps-final-report.png" caption="Updated ReactOS Application Manager" >}}
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:26]:
```suggestion
{{< figure link="/img/blogs/gsoc2020-rapps-final-report.png" src="/img/blogs/gsoc2020-rapps-final-report.png" caption="Updated ReactOS Applications Manager" >}}
```
See the window title on screenshot :)

File: content/project-news/reactos-in-2020.md [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+## Shell changes
+
+Shell hasn't seen much attention recently, due to most of the work being concentrated in the kernel, but there are still some useful fixes and feature implementations.
+
+* Katayama Hirofumi MZ made a great job implementing the file system change notification feature (aka `SHChangeNotify`) in the shell. This is required for all applications monitoring the filesystem changes.
+* Visual and usability improvements: "Size on disk" label in File Properties dialog, key accelerators in Open/Save dialog and many more.
+
+## RAPPS (a GSoC project)
+
+This is one of our three GSoC projects of the year. He Yang (kernel.bin) continues on the work made by GSoC students in previous years, by improving our RAPPS application manager. This includes UI & usability enhancements, showing app screenshots and command line scripting support.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:23]:
```suggestion
This is one of our three GSoC projects of the year. He Yang (kernel.bin) continues on the work made by GSoC students in previous years, by improving our RAPPS applications manager. This includes UI & usability enhancements, showing app screenshots and command line scripting support.
```

File: content/project-news/reactos-in-2020.md [hunk 35]
```diff
@@ -0,0 +1,108 @@
...
+as per evidence with the many failures encountered during testing, with the `cmd:batch` Wine unit test.
+
+{{< figure link="/img/project-news/reactos-in-2020/cmd-tests-changes.png" src="/img/project-news/reactos-in-2020/cmd-tests-changes.png" caption="Old test results can be seen on the left" >}}
+
+Hermès Bélusca-Maïto decided to make ReactOS' CMD parsing rules more compatible with Windows' one, including some obscure corner cases,
+and to take also the opportunity to bring additional fixes to some of its commands with the aim to fix more tests.
+
+Now, more `cmd:batch` tests are passing with overall less failures. He also added new tests for the CMD parser in our own `cmd:reactos` unit test. Most of these pass OK, except for some for which some extra work is still needed.
+
+Hermès has also found that some aspecss of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md]: Typo
```suggestion
Hermès has also found that some aspects of the implementation of ReactOS' existing CMD `copy` command, as well as the execution of piped external commands, do fail in certain cases. More work will be required in these areas in the future.
```

File: content/project-news/reactos-in-2020.md [hunk 36]
```diff
@@ -0,0 +1,108 @@
...
+## Ports
+
+Good old i386 still remains as our main platform, but the work on porting ReactOS to other architectures slowly goes on.
+
+* AMD64: most of the code was merged to master, so now nightly builds can finally boot to the 1st stage setup screen on some hardware. The graphical interface still needs fixes. The porting is being done by Timo Kreuzer.
+* XBOX: the Original XBOX port of ReactOS is revived, and able to boot to desktop. The work is mostly done by Stanislav Motylkov.
+* PC98: Dmitry Borisov (disean) started a NEC PC-98 port of ReactOS, which is a i386 flavor but with some incompatibilities.
+
+{{< gallery >}}
+{{< figure link="/img/project-news/reactos-in-2020/reactos-xbox.jpg" caption="ReactOS running on a real Xbox Original" >}}
```
Comment (binarymaster) [inline content/project-news/reactos-in-2020.md:97]: Better naming
```suggestion
{{< figure link="/img/project-news/reactos-in-2020/reactos-xbox.jpg" caption="ReactOS running on a real Original Xbox" >}}
```
